architecture/2024-08-resource-types-registration.md

```diff
@@ -0,0 +1,641 @@
+| --------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| Resource Provider           | A microservice that implements support for one or more resource types grouped under the same namespace (eg: Applications.Core). Radius provides several resource providers in the default installation. Resource provider can also refer to a group of related types under a common namespace. This proposal introduces an API for registering and unregistering resource providers. |
+| Resource Provider Namespace | A group of related or similar resources that share a resource provider. eg: In `Applications.Core/containers` the resource provider namespace is `Applications.Core`.                                                                                                                                                                                                                |
+
+| Resource Type      | An API entity that provides CRUDL operations. Resource Types are implemented by a resource provider and belong to a resource provider namespace. Resource types are a new resource type introduced by this proposal. This proposal introduces an API for registering and unregistering resource resource types.                                                                                                                                     |
+| OpenAPI/Schema     | A format for documenting HTTP-based APIs. In this document we're primarily concerned with the parts of OpenAPI pertaining to request/response bodies. |
+| Location/Region | An instance of a loosely-connected deployment of UCP. Regions of a UCP can collaborate to serve route requests between each other. A multi-regional deployment of UCP provides fault-isolation. In a multi-regional deployment of UCP, each region is responsible for processing requests that correspond to its region based on the `location` field. |
+
+## Objectives
+
+> **Issue Reference:** https://github.com/radius-project/radius/issues/6688
+
+### Goals
+
+- Support granular registration of resource providers and resource types (add/change/remove) by users.
+- Support efficient querying of resource type data for any API consumer.
+- Register resource types for existing Radius functionality like Deployment Engine and `Applications.*` resource types.
+- Support the future of UCP as a multi-regional control-plane by avoiding breaking changes in the future.
+  - For now we ensure the `location` field and concept is surfaced in the API by assuming that multi-regional deployments are possible. In practice only a single region will be present.
```

Comment:
it defaults to `global` today, correct?

Reply:
`global` is the sentinel we currently use to mean "there is only a single region". 

This is also a default we set in Bicep. The API doesn't have a default, it's just an API.